packages/ts-client/src/webrtc/webRTCEndpoint.ts

```diff
@@ -1607,13 +1588,16 @@ export class WebRTCEndpoint<
       const mid = event.transceiver.mid!;
 
       const trackId = this.midToTrackId.get(mid)!;
+
       if (this.checkIfTrackBelongToEndpoint(trackId, this.localEndpoint))
         return;
+      if (!isTrackKind(event.track.kind)) throw new Error("Track has no kind")
 
       const trackContext = this.trackIdToTrack.get(trackId)!;
 
       trackContext.stream = stream;
       trackContext.track = event.track;
+      trackContext.trackKind = event.track.kind;
```

Comment:
i may have missed something, but why are we copying the `event.track.kind` information to another field while we can access it through `trackCotnext.track.kind`?

Reply:
The `track` field is optional; it's possible to mute a track (replace the track with null) and then unmute it in the future. We need the `trackKind` to assess bandwidth even though the track is muted